DirectProgramming/C++SYCL/MapReduce/guided_MonteCarloMultiGPU_SYCLMigration/sample.json

```diff
@@ -14,9 +14,13 @@
 			"mkdir build",
       		        "cd build",
            		"cmake ..",
+			"cmake -D INTEL_MAX_GPU=1 ..",
+			"cmake -D NVIDIA_GPU=1 ..",
            		"make",
 			"make run_sm_cpu",

```

Comment:
@AvijitBag07 Getting this error message:

make run_sm_cpu
/home/u105196/pr1891/oneAPI-samples/DirectProgramming/C++SYCL/MapReduce/guided_MonteCarloMultiGPU_SYCLMigration/build/bin/02_sycl_migrated Starting...

Using single CPU thread for multiple GPUs
MonteCarloMultiGPU
==================
Parallelization method  = streamed
Problem scaling         = weak
Number of GPUs          = 1
Total number of options = 8192
Number of paths         = 262144
main(): generating input data...
main(): starting 1 host threads...
terminate called after throwing an instance of 'sycl::_V1::runtime_error'
  what():  Native API failed. Native API returns: -42 (PI_ERROR_INVALID_BINARY) -42 (PI_ERROR_INVALID_BINARY)
Aborted (core dumped)
make[3]: *** [02_sycl_migrated/CMakeFiles/run_sm_cpu.dir/build.make:70: 02_sycl_migrated/CMakeFiles/run_sm_cpu] Error 134
make[2]: *** [CMakeFiles/Makefile2:128: 02_sycl_migrated/CMakeFiles/run_sm_cpu.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:135: 02_sycl_migrated/CMakeFiles/run_sm_cpu.dir/rule] Error 2

Reply:
I have checked it is working for me.

- Plz check sycl-ls is opencl:cpu is available or not 

```
  sycl-ls
[opencl:cpu:0] Intel(R) OpenCL, Intel(R) Xeon(R) Gold 6128 CPU @ 3.40GHz 3.0 [2023.16.7.0.21_160000]
[opencl:acc:1] Intel(R) FPGA Emulation Platform for OpenCL(TM), Intel(R) FPGA Emulation Device 1.2 [2023.16.7.0.21_160000]
```
- Please follow these steps to run on CPU

```
u197045@s001-n017:~/Avijit/PR/MonteCarlo/oneAPI-samples/DirectProgramming/C++SYCL/MapReduce/guided_MonteCarloMultiGPU_SYCLMigration$ mkdir build
u197045@s001-n017:~/Avijit/PR/MonteCarlo/oneAPI-samples/DirectProgramming/C++SYCL/MapReduce/guided_MonteCarloMultiGPU_SYCLMigration$ cd build
u197045@s001-n017:~/Avijit/PR/MonteCarlo/oneAPI-samples/DirectProgramming/C++SYCL/MapReduce/guided_MonteCarloMultiGPU_SYCLMigration/build$ ls
u197045@s001-n017:~/Avijit/PR/MonteCarlo/oneAPI-samples/DirectProgramming/C++SYCL/MapReduce/guided_MonteCarloMultiGPU_SYCLMigration/build$ cmake ..
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is Clang 17.0.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /glob/development-tools/versions/oneapi/2023.2.0.1/oneapi/compiler/2023.2.1/linux/bin/icpx
-- Check for working CXX compiler: /glob/development-tools/versions/oneapi/2023.2.0.1/oneapi/compiler/2023.2.1/linux/bin/icpx -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Default CMAKE_BUILD_TYPE not set using Release with Debug Info
-- Configuring done
-- Generating done
-- Build files have been written to: /home/u197045/Avijit/PR/MonteCarlo/oneAPI-samples/DirectProgramming/C++SYCL/MapReduce/guided_MonteCarloMultiGPU_SYCLMigration/build
u197045@s001-n017:~/Avijit/PR/MonteCarlo/oneAPI-samples/DirectProgramming/C++SYCL/MapReduce/guided_MonteCarloMultiGPU_SYCLMigration/build$ make 
Scanning dependencies of target 02_sycl_migrated
[ 20%] Building CXX object 02_sycl_migrated/CMakeFiles/02_sycl_migrated.dir/Samples/5_Domain_Specific/MonteCarloMultiGPU/MonteCarloMultiGPU.cpp.dp.cpp.o
[ 40%] Building CXX object 02_sycl_migrated/CMakeFiles/02_sycl_migrated.dir/Samples/5_Domain_Specific/MonteCarloMultiGPU/MonteCarlo_gold.cpp.dp.cpp.o
[ 60%] Building CXX object 02_sycl_migrated/CMakeFiles/02_sycl_migrated.dir/Samples/5_Domain_Specific/MonteCarloMultiGPU/MonteCarlo_kernel.dp.cpp.o
[ 80%] Building CXX object 02_sycl_migrated/CMakeFiles/02_sycl_migrated.dir/Samples/5_Domain_Specific/MonteCarloMultiGPU/multithreading.cpp.o
[100%] Linking CXX executable ../bin/02_sycl_migrated
[100%] Built target 02_sycl_migrated
u197045@s001-n017:~/Avijit/PR/MonteCarlo/oneAPI-samples/DirectProgramming/C++SYCL/MapReduce/guided_MonteCarloMultiGPU_SYCLMigration/build$ make run_sm_cpu
Scanning dependencies of target run_sm_cpu
/home/u197045/Avijit/PR/MonteCarlo/oneAPI-samples/DirectProgramming/C++SYCL/MapReduce/guided_MonteCarloMultiGPU_SYCLMigration/build/bin/02_sycl_migrated Starting...

Using single CPU thread for multiple GPUs
MonteCarloMultiGPU
==================
Parallelization method  = streamed
Problem scaling         = weak
Number of GPUs          = 1
Total number of options = 8192
Number of paths         = 262144
main(): generating input data...
main(): starting 1 host threads...
main(): GPU statistics, streamed
GPU Device #0: Intel(R) Xeon(R) Gold 6128 CPU @ 3.40GHz
Options         : 8192
Simulation paths: 262144

Total time (ms.): 55606.726562
        Note: This is elapsed time for all to compute.
Options per sec.: 147.320306
main(): comparing Monte Carlo and Black-Scholes results...
Shutting down...
Test Summary...
L1 norm        : 4.791477E-04
Average reserve: 10.923597

NOTE: The CUDA Samples are not meant for performance measurements. Results may vary when GPU Boost is enabled.

Test passed
Built target run_sm_cpu
```